torch/quantization/fx/_equalize.py

```diff
@@ -357,6 +366,40 @@ def scale_weight_node(
     scaled_bias = torch.mul(bias, next_equalization_scale)
     modules[node.target].bias = nn.Parameter(scaled_bias)
 
+def scale_weight_functional(node: Node, model: GraphModule, equalization_scale: torch.Tensor):
+    """ Scales the weight value for functional layers
+    """
+
+    # Find the next functional node so that we can construct the weight observer nodes
+    while node.op != 'output' and node.op != 'call_function':
+        node = node.next
+
+    if node.op == 'output':
+        return
+
+    assert(isinstance(node.args[1], Node))
+    weight_observer_nodes = collect_producer_nodes(node.args[1])
+    if weight_observer_nodes is None:
+        return
+
+    for weight_node in weight_observer_nodes:
+        # Find the node containing the weight values
+        if weight_node.op == 'get_attr':
+            assert(isinstance(weight_node.target, str))
+            weight = model.get_buffer(weight_node.target)
+
+            # Scale the weights for input-weight equalization
+            # If the following layer needs to be equalized then we will multiply its scale
+            scaled_weight = torch.mul(weight, torch.reciprocal(equalization_scale))
+            # TODO: connecting functional layers?
+            # scaled_weight = torch.mul(scaled_weight, next_equalization_scale)
+
+            # Set the weight with the newly scaled weight in the original model
+            # weight_node.target looks something like "linear1.w", so we want to
+            # get rid of the last two characters to retrieve the path to the
+            # submodule containing the weight
+            model.get_submodule(weight_node.target[:-2]).w = scaled_weight
```

Comment:
This seems like it may fail if the user decides to use `.weight` instead of `.w` for the weight parameter.
Can we instead use `getattr` on the functional linear inputs (since we know input 1 is weight) and call `setattr` on it to update the weight?

Reply:
Good point, let me try out your suggestion!

Reply:
Jerry suggested using the `_parent_name()` function, so I have updated the code to be:
```python
parent_name, name = _parent_name(weight_node.target)
setattr(modules[parent_name], name, scaled_weight)
```